src/test/java/org/worshipsongs/importer/SongParserTest.java

```diff
@@ -93,4 +135,97 @@ public void testParseVerseOrder()
                 "verseOrder=V1 C1 O1 V2 V3\n" +
                 "foofoo"));
     }
+
+    @Test
+    public void testParseLyrics()
+    {
+        assertEquals(lyrics, parser.parseLyrics(lyrics +
+                "\ntitle="));
+        assertEquals(lyrics, parser.parseLyrics(lyrics +
+                "\nauthor="));
+        assertEquals(lyrics, parser.parseLyrics(lyrics));
+        assertEquals("", parser.parseLyrics(""));
+    }
+
+    @Test
+    public void testParseSearchLyrics()
+    {
+        assertEquals("", parser.parseSearchLyrics(""));
+        assertEquals(searchLyrics, parser.parseSearchLyrics(lyrics));
+    }
+
+    @Test
+    public void testGetXmlLyrics()
+    {
+        String expectedLyrics = "";
+        ClassLoader classLoader = getClass().getClassLoader();
+        try {
+            expectedLyrics = IOUtils.toString(classLoader.getResourceAsStream("sample_lyrics.xml"));
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+        }
+        assertEquals(expectedLyrics.toString(), parser.getXmlLyrics(lyrics, "v1 o1 c1 o2 o3"));
+    }
+
+    @Test
+    public void testGetVerseTag() throws TransformerException
+    {
+        Element verseTag = parser.getVerseTag(document);
+        document.appendChild(verseTag);
+        transformer.setOutputProperty("omit-xml-declaration", "yes");
+        transformer.transform(new DOMSource(document), new StreamResult(out));
+        assertEquals("<verse type=\"\" label=\"\"><![CDATA[data]]></verse>", out.toString());
+    }
+
+    @Test
+    public void testGetLyricsTag() throws TransformerException
+    {
+        Element lyricsTag = parser.getLyricsTag(document);
+        document.appendChild(lyricsTag);
+        transformer.setOutputProperty("omit-xml-declaration", "yes");
+        transformer.transform(new DOMSource(document), new StreamResult(out));
+        assertEquals("<lyrics/>", out.toString());
+    }
+
+    @Test
+    public void testGetSongTag() throws TransformerException
+    {
+        document.setXmlStandalone(true);
+        Element songTag = parser.getSongTag(document);
+        document.appendChild(songTag);
+        transformer.transform(new DOMSource(document), new StreamResult(out));
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\"?><song version=\"1.0\"/>", out.toString());
+    }
+
+    @Test
+    public void testParseVerseOrders()
+    {
+        assertEquals("V1", parser.parseVerseOrders("V1 V2 V3 V4")[0]);
+        assertEquals("V2", parser.parseVerseOrders("V1 V2 V3 V4")[1]);
+        assertEquals("V3", parser.parseVerseOrders("V1 V2 V3 V4")[2]);
+        assertEquals("V4", parser.parseVerseOrders("V1 V2 V3 V4")[3]);
+    }
+
+    @Test
+    public void testParseVerseType()
+    {
+        assertEquals("V", parser.parseVerseType("V1"));
```

Comment:
Please verify from OpenLP db whether the verse type should be lowercase or not. Make appropriate changes accordingly.